src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/ItemCooldowns.java

```diff
@@ -153,36 +178,78 @@ private static boolean isPickaxe(String internalname) {
 		} else return internalname.equals("GEMSTONE_GAUNTLET") || internalname.equals("PICKONIMBUS") || internalname.equals("DIVAN_DRILL");
 	}
 
-	private static void updatePickaxeCooldown() {
-		if (pickaxeCooldown == -1 && NotEnoughUpdates.INSTANCE.config.itemOverlays.pickaxeAbility) {
-			for (ItemStack stack : Minecraft.getMinecraft().thePlayer.inventory.mainInventory) {
-				if (stack != null && stack.hasTagCompound()) {
-					String internalname = NotEnoughUpdates.INSTANCE.manager.getInternalNameForItem(stack);
-					if (isPickaxe(internalname)) {
-						for (String line : NotEnoughUpdates.INSTANCE.manager.getLoreFromNBT(stack.getTagCompound())) {
-							Matcher matcher = PICKAXE_COOLDOWN_LORE_REGEX.matcher(line);
-							if (matcher.find()) {
-								try {
-									pickaxeCooldown = Integer.parseInt(matcher.group(1));
-									return;
-								} catch (Exception ignored) {
-								}
-							}
-						}
-					}
+	@SubscribeEvent
+	public void onChatMessage(ClientChatReceivedEvent event) {
+		if (PICKAXE_ABILITY_ACTIVATION.matcher(event.message.getFormattedText()).matches() &&
+			NotEnoughUpdates.INSTANCE.config.itemOverlays.pickaxeAbility && pickaxeCooldown != 0) {
+			findCooldownInTooltip("Pickaxe");
+			pickaxeUseCooldownMillisRemaining = pickaxeCooldown * 1000;
+		}
+
+		if (BONZO_ABILITY_ACTIVATION.matcher(event.message.getFormattedText()).matches() &&
+				NotEnoughUpdates.INSTANCE.config.itemOverlays.bonzoAbility &&
+				(bonzoMaskCooldown != 0 || fraggedBonzoMaskCooldown != 0)) {
+			if (event.message.getFormattedText().contains("⚚")) {
+				findCooldownInTooltip("Fragged Bonzo Mask");
+				fraggedBonzomaskCooldownMillisRemaining = fraggedBonzoMaskCooldown * 1000;
+			} else {
+				findCooldownInTooltip("Bonzo Mask");
+				bonzomaskCooldownMillisRemaining = bonzoMaskCooldown * 1000;
+			}
+		}
+
+		if (SPIRIT_ABILITY_ACTIVATION.matcher(event.message.getFormattedText()).matches() &&
+				NotEnoughUpdates.INSTANCE.config.itemOverlays.spiritAbility && spiritMaskCooldown != 0) {
+			findCooldownInTooltip("Spirit Mask");
+			spiritMaskCooldownMillisRemaining = spiritMaskCooldown * 1000;
+		}
+	}
+
+	private static void findCooldownInTooltip(String item) {
+		for (ItemStack stack : Minecraft.getMinecraft().thePlayer.inventory.mainInventory) {
+			setSpecificCooldown(stack, item);
+		}
+
+		// Check helmet slot for items that can also be equipped as a helmet
+		ItemStack stack = Minecraft.getMinecraft().thePlayer.inventory.armorInventory[3];
+		setSpecificCooldown(stack, item);
+	}
+
+	private static void setSpecificCooldown(ItemStack stack, String item) {
+		if (stack != null && stack.hasTagCompound()) {
+
+			String internalname = NotEnoughUpdates.INSTANCE.manager.createItemResolutionQuery().withItemStack(stack).resolveInternalName();
+
+			if (internalname != null) {
+				switch (item) {
+					case "Pickaxe":
```

Comment:
Not a big fan of using magic strings like this. Maybe use an enum, or extract the strings into constants. Other than that great PR

Reply:
I thought strings wasn't quite the correct thing at the time, and forgot about enums until you commented this so I've changed them to enums now.